core-web/apps/dotcms-ui/src/app/portlets/dot-containers/dot-container-create/dot-container-code/dot-add-variable/store/dot-add-variable.store.ts

```diff
@@ -7,48 +7,76 @@ import { Injectable } from '@angular/core';
 
 import { DotGlobalMessageService } from '@components/_common/dot-global-message/dot-global-message.service';
-import { DotContentTypeService } from '@dotcms/data-access';
+import { DotContentTypeService, DotMessageService } from '@dotcms/data-access';
 import { DotCMSContentType, DotCMSContentTypeField } from '@dotcms/dotcms-models';
 import { DotHttpErrorManagerService } from '@services/dot-http-error-manager/dot-http-error-manager.service';
 
+import { DotFieldContent, FilteredFieldTypes } from '../dot-add-variable.models';
+
 export interface DotAddVariableState {
-    variables: DotCMSContentTypeField[];
+    fields: DotFieldContent[];
 }
 
 @Injectable()
 export class DotAddVariableStore extends ComponentStore<DotAddVariableState> {
     constructor(
         private dotContentTypeService: DotContentTypeService,
         private dotGlobalMessageService: DotGlobalMessageService,
-        private dotHttpErrorManagerService: DotHttpErrorManagerService
+        private dotHttpErrorManagerService: DotHttpErrorManagerService,
+        private dotMessage: DotMessageService
     ) {
         super({
-            variables: []
+            fields: []
         });
     }
 
-    readonly vm$ = this.select(({ variables }) => {
+    readonly vm$ = this.select(({ fields }) => {
         return {
-            variables
+            fields
         };
     });
 
-    readonly updateVariables = this.updater<DotCMSContentTypeField[]>(
-        (state: DotAddVariableState, variables: DotCMSContentTypeField[]) => {
+    readonly updateFields = this.updater<DotCMSContentTypeField[]>(
+        (state: DotAddVariableState, fields: DotCMSContentTypeField[]) => {
             return {
                 ...state,
-                variables
+                fields: fields.reduce(
+                    (acc, field) => {
+                        const { fieldType } = field;
+
+                        // If you want to filter a new field type, add it to the FilteredFieldTypes enum
+                        if ((Object.values(FilteredFieldTypes) as string[]).includes(fieldType)) {
+                            return acc;
+                        }
+
+                        acc.push(
+                            // This try to find the fields by field type, if it doesn't exist it will use the default one
+                            ...(this.fields[fieldType]?.(field) ?? this.fields.default(field))
+                        );
+
+                        return acc;
+                    },
+                    [
+                        // We initialize the array with the Content Identifier field
+                        {
+                            name: this.dotMessage.get('Content-Identifier-value'),
+                            variable: 'ContentIdentifier',
+                            fieldTypeLabel: this.dotMessage.get('Content-Identifier'),
+                            codeTemplate: this.getCodeTemplate.default('ContentIdentifier')
+                        }
```

Comment:
This can be a constant:

```ts
private readonly CONTENT_IDENTIFIER_FIELD: DotFieldContent = {
                            name: this.dotMessage.get('Content-Identifier-value'),
                            variable: 'ContentIdentifier',
                            fieldTypeLabel: this.dotMessage.get('Content-Identifier'),
                            codeTemplate: this.getCodeTemplate.default('ContentIdentifier')
                        }

```